ui/src/app/machines/views/MachineDetails/MachineNetwork/EditPhysicalForm/EditPhysicalForm.test.tsx

```diff
@@ -0,0 +1,192 @@
+        items: [subnetFactory(), subnetFactory()],
+        loaded: true,
+      }),
+      vlan: vlanStateFactory({
+        items: [vlanFactory(), vlanFactory()],
+        loaded: true,
+      }),
+    });
+  });
+
+  it("fetches the necessary data on load", () => {
+    const store = mockStore(state);
+    mount(
+      <Provider store={store}>
+        <MemoryRouter
+          initialEntries={[{ pathname: "/machines", key: "testKey" }]}
+        >
+          <EditPhysicalForm nicId={1} systemId="abc123" close={jest.fn()} />
+        </MemoryRouter>
+      </Provider>
+    );
+    const expectedActions = ["fabric/fetch", "vlan/fetch"];
+    expectedActions.forEach((expectedAction) => {
+      expect(
+        store.getActions().some((action) => action.type === expectedAction)
+      );
+    });
+  });
+
+  it("displays a spinner when data is loading", () => {
+    state.vlan.loaded = false;
+    state.fabric.loaded = false;
+    const store = mockStore(state);
+    const wrapper = mount(
+      <Provider store={store}>
+        <MemoryRouter
+          initialEntries={[{ pathname: "/machines", key: "testKey" }]}
+        >
+          <EditPhysicalForm nicId={1} systemId="abc123" close={jest.fn()} />
+        </MemoryRouter>
+      </Provider>
+    );
+    expect(wrapper.find("Spinner").exists()).toBe(true);
+  });
+
+  it("does not allow the link speed to be higher than the interface speed", async () => {
+    const store = mockStore(state);
+    const wrapper = mount(
+      <Provider store={store}>
+        <MemoryRouter
+          initialEntries={[{ pathname: "/machines", key: "testKey" }]}
+        >
+          <EditPhysicalForm nicId={1} systemId="abc123" close={jest.fn()} />
+        </MemoryRouter>
+      </Provider>
+    );
+    await act(async () => {
+      wrapper.find("input[name='interface_speed']").simulate("change", {
+        target: {
+          name: "interface_speed",
+          value: 1,
+        },
+      });
+    });
+    wrapper.update();
+    await act(async () => {
+      wrapper.find("input[name='link_speed']").simulate("change", {
+        target: {
+          name: "link_speed",
+          value: 2,
+        },
+      });
+    });
+    await act(async () => {
+      wrapper.find("input[name='link_speed']").simulate("blur");
+    });
+    wrapper.update();
+    expect(wrapper.find(".p-form-validation__message").exists()).toBe(true);
+    expect(wrapper.find(".p-form-validation__message").text()).toBe(
+      "Error: Link speed cannot be higher than interface speed"
+    );
+  });
+
+  it("correctly dispatches actions to edit a physical interface", () => {
+    const store = mockStore(state);
+    const wrapper = mount(
+      <Provider store={store}>
+        <MemoryRouter
+          initialEntries={[{ pathname: "/machines", key: "testKey" }]}
+        >
+          <EditPhysicalForm nicId={1} systemId="abc123" close={jest.fn()} />
+        </MemoryRouter>
+      </Provider>
+    );
+
+    act(() =>
```

Comment:
Does the new 'wait' function not work here?

Reply:
It does, thanks.